src/utils/hooks/useTriggerProcessing.tsx

```diff
@@ -3,32 +3,46 @@ import { useGetTriggersQuery } from '../../utils/generated/graphql'
 
 type TriggerType = 'applicationTrigger' | 'reviewAssignmentTrigger' | 'reviewTrigger'
 
-interface TriggerQueryProps {
-  checkTrigger: boolean
-  serialNumber?: string
-  reviewAssignmentId?: number
-  reviewId?: number
-  triggerType?: TriggerType
+interface TriggerApplication {
+  isApplicationLoaded: boolean
+  serialNumber: string
 }
+interface TriggerReview {
+  isReviewLoaded: boolean
+  reviewId: number
+}
+interface TriggerAssignment {
+  reviewAssignmentId: number
+}
+type TriggerQueryProps = { triggerType?: TriggerType } & (
+  | TriggerApplication
+  | TriggerReview
+  | TriggerAssignment
+)
```

Comment:
Nice :) 
So this combines TriggerType with one of the specific types (app, review, etc). I guess it's not really feasible to ensure that the OR types match the TriggerType?

Reply:
I think we can use it at some point, I just wasn't sure how to do, so left it not using it to deduce. Let's improve in a next PR